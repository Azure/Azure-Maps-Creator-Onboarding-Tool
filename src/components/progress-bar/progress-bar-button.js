import { cx } from '@emotion/css';
import { ActionButton } from '@fluentui/react';
import { useCompletedSteps } from 'common/store';
import { useCustomNavigate } from 'hooks';
import PropTypes from 'prop-types';
import { useCallback, useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { useLocation } from 'react-router-dom';
import ProgressBarIcon from './progress-bar-icon';
import { activeStepStyle, disabledStepStyle, stepStyle } from './progress-bar.style';

const ProgressBarButton = ({ step }) => {
  const completedSteps = useCompletedSteps();
  const { t } = useTranslation();
  const { pathname } = useLocation();
  const navigate = useCustomNavigate();

  const isActiveStep = useMemo(() => pathname === step.href, [pathname, step]);
  const isCompletedStep = useMemo(() => completedSteps.includes(step.key), [completedSteps, step]);
  const isDisabledStep = step.disabled;

  const onClick = useCallback(() => {
    navigate(step.href);
  }, [navigate, step]);

  return (
    <ActionButton
      className={cx(stepStyle, { [activeStepStyle]: isActiveStep, [disabledStepStyle]: isDisabledStep })}
      key={step.key}
      disabled={isActiveStep || isDisabledStep}
      onClick={onClick}
    >
      <ProgressBarIcon label={t(step.name)} isCompletedStep={isCompletedStep} iconName={step.icon} />
      {t(step.name)}
    </ActionButton>
  );
};

ProgressBarButton.propTypes = {
  step: PropTypes.shape({
    href: PropTypes.string.isRequired,
    key: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
};

export default ProgressBarButton;
