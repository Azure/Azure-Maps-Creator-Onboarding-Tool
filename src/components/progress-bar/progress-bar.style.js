import { css } from '@emotion/css';
import { color, fontSize, fontWeight } from 'common/styles';

export const errorContainer = css`
  max-width: 31.25rem;
  margin: 0.9375rem 0;
  display: flex;
  flex-direction: column;
  gap: 1rem;
`;

export const progressBarContainer = css`
  margin-bottom: 1rem;
  display: flex;
  gap: 0.5rem;
`;

export const iconStyle = css`
  color: ${color.gray};
  font-size: ${fontSize.xl};
  margin-right: 0.5rem;
`;

export const iconCompletedStyle = css`
  color: ${color.greenIcon};
`;

export const iconErrorStyle = css`
  color: ${color.redError};
`;

export const stepStyle = css`
  font-size: ${fontSize.md};
  height: 2.25rem;
  display: inline-flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
  padding: 0.3rem;
  transition: background 0.1s ease-in;
  &:hover {
    color: inherit;
    background-color: ${color.lightGray};
  }
`;

export const activeStepStyle = css`
  font-weight: ${fontWeight.semibold};
  &::after {
    position: absolute;
    content: '';
    display: block;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 2px;
    background-color: ${color.accent.primary};
  }
`;

export const disabledStepStyle = css`
  border-bottom: none;
  opacity: 0.5;
  cursor: not-allowed;
`;
