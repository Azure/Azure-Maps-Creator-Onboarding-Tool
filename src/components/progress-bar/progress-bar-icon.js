import PropTypes from 'prop-types';
import { cx } from '@emotion/css';
import { Icon } from '@fluentui/react/lib/Icon';

import { iconCompletedStyle, iconErrorStyle, iconStyle } from './progress-bar.style';
import { useProgressBarStore } from 'common/store';

const progressBarSelector = (s) => s.isErrorShown;

const ProgressBarIcon = ({isCompletedStep, label}) => {
  const isProgressBarErrorShown = useProgressBarStore(progressBarSelector);

  if (isProgressBarErrorShown && !isCompletedStep) {
    return (
      <Icon iconName='ErrorBadge' ariaLabel={label} className={cx(iconStyle, iconErrorStyle)} />
    );
  }

  return (
    <Icon iconName='SkypeCircleCheck' ariaLabel={label}
          className={cx(iconStyle, { [iconCompletedStyle]: isCompletedStep })}
    />
  );
};

ProgressBarIcon.propTypes = {
  isCompletedStep: PropTypes.bool.isRequired,
  label: PropTypes.string.isRequired,
};

export default ProgressBarIcon;