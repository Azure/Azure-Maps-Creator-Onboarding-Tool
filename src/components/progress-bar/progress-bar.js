import { MessageBar, MessageBarType } from '@fluentui/react';
import { useCompletedSteps, useProgressBarSteps, useProgressBarStore } from 'common/store';
import { useEffect, useMemo } from 'react';
import { getFeatureFlags } from 'utils';
import { useTranslation } from 'react-i18next';
import { useLocation } from 'react-router-dom';
import ProgressBarButton from './progress-bar-button';
import { errorContainer, progressBarContainer } from './progress-bar.style';

const progressBarStoreSelector = s => [
  s.isIncorrectManifestVersionErrorShown,
  s.hideIncorrectManifestVersionError,
  s.isInvalidManifestErrorShown,
  s.hideInvalidManifestError,
  s.isMissingDataErrorShown,
  s.hideMissingDataError,
];

export const ProgressBar = () => {
  const { t } = useTranslation();
  const { pathname } = useLocation();
  const { isPlacesPreview } = getFeatureFlags();
  const completedSteps = useCompletedSteps();
  const [
    isIncorrectManifestVersionErrorShown,
    hideIncorrectManifestVersionError,
    isInvalidManifestErrorShown,
    hideInvalidManifestError,
    isMissingDataErrorShown,
    hideMissingDataError,
  ] = useProgressBarStore(progressBarStoreSelector);

  const progressBarSteps = useProgressBarSteps();

  const shouldShowProgressBar = useMemo(
    () => progressBarSteps.some(step => step.href === pathname),
    [pathname, progressBarSteps]
  );

  useEffect(() => {
    if (completedSteps.length === progressBarSteps.length && isMissingDataErrorShown) {
      hideMissingDataError();
    }
  }, [completedSteps, isMissingDataErrorShown, hideMissingDataError, progressBarSteps]);

  if (!shouldShowProgressBar) {
    return null;
  }

  return (
    <>
      <div className={errorContainer}>
        {isIncorrectManifestVersionErrorShown && !isPlacesPreview &&(
          <MessageBar messageBarType={MessageBarType.warning} isMultiline onDismiss={hideIncorrectManifestVersionError}>
            {t('error.manifest.incorrect.version')}
          </MessageBar>
        )}
        {isIncorrectManifestVersionErrorShown && isPlacesPreview &&(
          <MessageBar messageBarType={MessageBarType.warning} isMultiline onDismiss={hideIncorrectManifestVersionError}>
            {t('error.manifest.places.incorrect.version')}
          </MessageBar>
        )}
        {isInvalidManifestErrorShown && (
          <MessageBar messageBarType={MessageBarType.warning} isMultiline onDismiss={hideInvalidManifestError}>
            {t('error.manifest.invalid')}
          </MessageBar>
        )}
        {isMissingDataErrorShown && (
          <MessageBar messageBarType={MessageBarType.error} isMultiline>
            {t('error.validation.failed.missing.info')}
          </MessageBar>
        )}
      </div>
      <div className={progressBarContainer}>
        {progressBarSteps.map(step => (
          <ProgressBarButton step={step} key={step.key} />
        ))}
      </div>
    </>
  );
};

export default ProgressBar;
