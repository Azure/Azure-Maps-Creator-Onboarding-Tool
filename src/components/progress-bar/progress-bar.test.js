import { fireEvent, render, screen } from '@testing-library/react';

import { ProgressBar } from './progress-bar';

const mockNavigate = jest.fn();
let mockCurrentPath;

jest.mock('react-router-dom', () => ({
  useLocation: () => ({
    pathname: mockCurrentPath,
  }),
  useNavigate: () => mockNavigate,
}));

describe('Progress Bar component', () => {
  beforeEach(() => {
    mockCurrentPath = '/levels';
  });

  it('should not render component', () => {
    mockCurrentPath = '/another/one/bites/the/dust';
    const view = render(<ProgressBar />);
    expect(view).toMatchSnapshot();
  });

  it('should render component', () => {
    const view = render(<ProgressBar />);
    expect(view).toMatchSnapshot();
  });

  it('should navigate to /layers when button is clicked', () => {
    render(<ProgressBar />);
    const button = screen.getByText('dwg.layers');

    fireEvent.click(button);
    expect(mockNavigate).toHaveBeenCalledWith('/layers');
  });
});