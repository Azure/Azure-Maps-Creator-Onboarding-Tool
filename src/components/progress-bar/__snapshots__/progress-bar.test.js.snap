// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Bar component should not render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Progress Bar component should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  max-width: 31.25rem;
  margin: 0.9375rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
}

.emotion-1 {
  margin-bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
}

.emotion-2 {
  font-size: 0.875rem;
  height: 2.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  padding: 0.3rem;
  -webkit-transition: background 0.1s ease-in;
  transition: background 0.1s ease-in;
  font-weight: 600;
}

.emotion-2:hover {
  color: inherit;
  background-color: #F3F2F1;
}

.emotion-2::after {
  position: absolute;
  content: '';
  display: block;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 2px;
  background-color: #0078d4;
}

.emotion-3 {
  color: #E1DFDD;
  font-size: 1rem;
  margin-right: 0.5rem;
}

.emotion-4 {
  font-size: 0.875rem;
  height: 2.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  padding: 0.3rem;
  -webkit-transition: background 0.1s ease-in;
  transition: background 0.1s ease-in;
}

.emotion-4:hover {
  color: inherit;
  background-color: #F3F2F1;
}

<body>
    <div>
      <div
        class="emotion-0"
      />
      <div
        class="emotion-1"
      >
        <button
          aria-disabled="true"
          class="ms-Button ms-Button--action ms-Button--command is-disabled emotion-2 root-109"
          data-is-focusable="false"
          disabled=""
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-110"
            data-automationid="splitbuttonprimary"
          >
            <i
              aria-label="facility.levels"
              class="emotion-3 root-119"
              data-icon-name="SkypeCircleCheck"
              role="img"
            >
              
            </i>
            facility.levels
          </span>
        </button>
        <button
          class="ms-Button ms-Button--action ms-Button--command emotion-4 root-120"
          data-is-focusable="true"
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-110"
            data-automationid="splitbuttonprimary"
          >
            <i
              aria-label="georeference"
              class="emotion-3 root-119"
              data-icon-name="SkypeCircleCheck"
              role="img"
            >
              
            </i>
            georeference
          </span>
        </button>
        <button
          class="ms-Button ms-Button--action ms-Button--command emotion-4 root-120"
          data-is-focusable="true"
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-110"
            data-automationid="splitbuttonprimary"
          >
            <i
              aria-label="dwg.layers"
              class="emotion-3 root-119"
              data-icon-name="SkypeCircleCheck"
              role="img"
            >
              
            </i>
            dwg.layers
          </span>
        </button>
        <button
          class="ms-Button ms-Button--action ms-Button--command emotion-4 root-120"
          data-is-focusable="true"
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-110"
            data-automationid="splitbuttonprimary"
          >
            <i
              aria-label="review.plus.create"
              class="emotion-3 root-119"
              data-icon-name="SkypeCircleCheck"
              role="img"
            >
              
            </i>
            review.plus.create
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  max-width: 31.25rem;
  margin: 0.9375rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
}

.emotion-1 {
  margin-bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
}

.emotion-2 {
  font-size: 0.875rem;
  height: 2.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  padding: 0.3rem;
  -webkit-transition: background 0.1s ease-in;
  transition: background 0.1s ease-in;
  font-weight: 600;
}

.emotion-2:hover {
  color: inherit;
  background-color: #F3F2F1;
}

.emotion-2::after {
  position: absolute;
  content: '';
  display: block;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 2px;
  background-color: #0078d4;
}

.emotion-3 {
  color: #E1DFDD;
  font-size: 1rem;
  margin-right: 0.5rem;
}

.emotion-4 {
  font-size: 0.875rem;
  height: 2.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  padding: 0.3rem;
  -webkit-transition: background 0.1s ease-in;
  transition: background 0.1s ease-in;
}

.emotion-4:hover {
  color: inherit;
  background-color: #F3F2F1;
}

<div>
    <div
      class="emotion-0"
    />
    <div
      class="emotion-1"
    >
      <button
        aria-disabled="true"
        class="ms-Button ms-Button--action ms-Button--command is-disabled emotion-2 root-109"
        data-is-focusable="false"
        disabled=""
        type="button"
      >
        <span
          class="ms-Button-flexContainer flexContainer-110"
          data-automationid="splitbuttonprimary"
        >
          <i
            aria-label="facility.levels"
            class="emotion-3 root-119"
            data-icon-name="SkypeCircleCheck"
            role="img"
          >
            
          </i>
          facility.levels
        </span>
      </button>
      <button
        class="ms-Button ms-Button--action ms-Button--command emotion-4 root-120"
        data-is-focusable="true"
        type="button"
      >
        <span
          class="ms-Button-flexContainer flexContainer-110"
          data-automationid="splitbuttonprimary"
        >
          <i
            aria-label="georeference"
            class="emotion-3 root-119"
            data-icon-name="SkypeCircleCheck"
            role="img"
          >
            
          </i>
          georeference
        </span>
      </button>
      <button
        class="ms-Button ms-Button--action ms-Button--command emotion-4 root-120"
        data-is-focusable="true"
        type="button"
      >
        <span
          class="ms-Button-flexContainer flexContainer-110"
          data-automationid="splitbuttonprimary"
        >
          <i
            aria-label="dwg.layers"
            class="emotion-3 root-119"
            data-icon-name="SkypeCircleCheck"
            role="img"
          >
            
          </i>
          dwg.layers
        </span>
      </button>
      <button
        class="ms-Button ms-Button--action ms-Button--command emotion-4 root-120"
        data-is-focusable="true"
        type="button"
      >
        <span
          class="ms-Button-flexContainer flexContainer-110"
          data-automationid="splitbuttonprimary"
        >
          <i
            aria-label="review.plus.create"
            class="emotion-3 root-119"
            data-icon-name="SkypeCircleCheck"
            role="img"
          >
            
          </i>
          review.plus.create
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
